lib/parallel_cucumber/runtime.js

```diff
@@ -195,33 +196,94 @@ var Runtime = function(configuration) {
     var tasks = [];
     var taskCount = 0;
 
-    options.featureFilePaths.forEach(function(featureFilePath) {
-      Object.keys(options.profiles).forEach(function(profileName) {
-        var taskIndex = taskCount++;
-        var profile = options.profiles[profileName];
-        // Clone the array using slice()
-        var tags = profile.tags ? profile.tags.slice() : [];
-        var env = profile.env || {};
-
-        env.PARALLEL_CUCUMBER_PROFILE = profileName;
-
-        var task = {
-          taskIndex: taskIndex,
-          profileName: profileName,
-          featureFilePath: featureFilePath,
-          supportCodePaths: options.supportCodePaths,
-          tags: tags,
-          env: env,
-          retryCount: 0
-        };
-
-        tasks.push(task);
+    options.featureFilePaths.forEach(function (featureFilePath) {
+      self._getFeatureFileObjects(featureFilePath).forEach(function (featureFile) {
+        Object.keys(options.profiles).forEach(function (profileName) {
+          var profile = options.profiles[profileName];
+          // Clone the array using slice()
+          var tags = profile.tags ? profile.tags.slice() : [];
+          var env = profile.env || {};
+
+          env.PARALLEL_CUCUMBER_PROFILE = profileName;
+
+          var matchTags = true;
+          tags.forEach(function (tag) {
+            if (matchTags) {
+              if (tag.indexOf('~') === 0) {
+                matchTags = featureFile.tags.indexOf(tag.substring(1)) === -1;
+              }
+              else {
+                matchTags = featureFile.tags.indexOf(tag) > -1;
+              }
+            }
+          });
+          if (matchTags) {
+            var task = {
+              taskIndex: taskCount++,
+              profileName: profileName,
+              featureFilePath: featureFile.path,
+              supportCodePaths: options.supportCodePaths,
+              tags: tags,
+              env: env,
+              retryCount: 0
+            };
+            tasks.push(task);
+          }
+        });
       });
     });
 
     callback(null, tasks);
   };
 
+  /**
+   * Get separate objects for every scenario and example in the given feature file.
+   * @param featureFilePath the feature file path
+   * @returns {Array} containing feature file objects
+   * @private
+   */
+  self._getFeatureFileObjects = function (featureFilePath) {
+    var featureFileObjects = [];
+    var featureTags = [];
+    var scenarioTags = [];
+    var tags = [];
+    var examplesHeader = false;
+
+    FS.readFileSync(featureFilePath, 'utf8').split(/[\n]/).forEach(function (line, i) {
+      if (new RegExp('@\\S+').test(line)) {
+        tags = tags.concat(line.match(new RegExp('(@\\S+)', 'g')));
+      }
+      else if (new RegExp('Feature:').test(line)) {
```

Comment:
@simondean I updated this part, using the gherkin library to parse the feature file. This is what cucumber uses itself, so it will ensure we do not miss anything.